File: CSVParser.go
package main

import (
    "fmt"
    "strings"
    "os"
    "bufio"
    "io"
    "bytes"
    "regexp"
    "strconv"
)

const filename string = "./testdata.txt"
func scrubQuote(s string) string {
  clean:=strings.Trim(s, " ")
  re1:=regexp.MustCompile(`\"`)
  clean = re1.ReplaceAllString(clean,"")
  re2:=regexp.MustCompile(`\\r\\n`)
  clean = re2.ReplaceAllString(clean,"")
  re3:=regexp.MustCompile(`\\n`)
  clean =re3.ReplaceAllString(clean,"")
  clean = strconv.Quote(clean)
  return clean
}
func formatOutput(records []map[string]interface{}) {
  var final []string
  var buffer bytes.Buffer
  for _, element := range records  {
    var pairs []string
    for k,v := range element {
      buffer.Reset()
      buffer.WriteString(k)
      buffer.WriteString(":")
      buffer.WriteString(v.(string))
      pairs = append(pairs,buffer.String())
    }
    rec:=strings.Join(pairs[:],",")
    buffer.Reset()
    buffer.WriteString("  {")
    buffer.WriteString(rec)
    buffer.WriteString("}")
    final = append(final,buffer.String())

  }
  recs:=strings.Join(final[:],",\n")
  fmt.Printf("{\n  \"payload\":[%s\n  ]\n}\n", recs)
}

func readFile() (err error) {
    file, err := os.Open(filename)
    defer file.Close()
    if err != nil {
        return err
    }
    // Start reading from the file with a reader.
    reader := bufio.NewReader(file)
    var records []map[string]interface{}
    records = make([]map[string]interface{},0,0)
    var count int
    var headers []string
    count=0
    for {
        var buffer bytes.Buffer
        var l []byte
        var isPrefix bool
        for {
            l, isPrefix, err = reader.ReadLine()
            buffer.Write(l)
            // EOL
            if !isPrefix {
                break
            }
            // EOF
            if err != nil {
                break
            }
        }
        if err == io.EOF {
            break
        }
        line := buffer.String()

        if count==0 { //header
          headers=strings.Split(line, ",")
          count++ //don't care after this
          continue
        } else {
          rec:=strings.Split(line, ",")
          var m map[string]interface{}
          m = make(map[string]interface{})
          for index, element := range headers {
            m[scrubQuote(element)]=scrubQuote(rec[index])
          }
          records = append(records, m)
        }
    }

    if err != io.EOF {
      fmt.Printf(" > Error : %v\n", err)
    }
    formatOutput(records)
    return
}
func main() {
  readFile()
}

File: CSVParser.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class CSVParser {

	Map<String,String> map = new HashMap<>();
	List<Map<String,String>> records = new ArrayList<>();
	/**
	 * Quick scrub
	 * @param s input strng
	 * @return clean string
	 */
	public String scrubQuote(String s) {
		String clean = s.trim();
		clean = clean.replaceAll("\"","");
		clean = clean.replaceAll("(.*)\\\\r\\\\n(.*)","$1$2");
		clean = clean.replaceAll("(.*)\\\\n(.*)","$1$2");
		return "\"" + clean + "\"";
	}
	/**
	 * slurp file in and scrub strings into lists, make maps
	 * @param fileName input file
	 * @throws IOException trouble reading file
	 */
	public void readFile(String fileName) throws IOException  {
		try {
			FileReader f = new FileReader(fileName);
			BufferedReader inStream = new BufferedReader(f);
	        String s;
	        List<String> header = null;
	        List<String> record = null;
	        int counter=0;
	        while ((s = inStream.readLine()) != null) {
	        	List<String> fields = (new ArrayList<>(Arrays.asList(s.split(",")))).stream().map( e -> scrubQuote(e)).collect(Collectors.toList());
	        	if (counter==0) {
	        		header = fields;
	        	} else {
	                record = fields;
	                map = new HashMap<>();
	                for (int i=0;i<header.size();i++) map.put(header.get(i),record.get(i));
	                records.add(map);
	        	}
	        	counter++;
	        }
	        inStream.close();
		} catch (FileNotFoundException e) {
			System.out.println("File Not Found");
			e.printStackTrace();
		}
		
	}
	
	public void dumpJSON() {
		System.out.println("{\n  \"payload\":[");
        List<String> payload = new ArrayList<>();
        records.stream().forEach((r) -> { 
        	StringBuffer rec = new StringBuffer();
        	r.entrySet().stream().forEach(e -> {
        		rec.append(e.getKey()).append(":").append(e.getValue()).append(",");
        	});
        	payload.add("{" + (rec.substring(0, rec.length()-1) + "}"));
        	rec.setLength(0);
        });
        String records = String.join(",\n",payload);
        System.out.println(records + "  ]\n}\n");
	}
	public static void main (String[] args)  {
		CSVParser csv = new CSVParser();
		try {
			csv.readFile("./testdata.txt");
			csv.dumpJSON();
		} catch (IOException e) {
			System.out.println("Error Reading File");
			e.printStackTrace();
		}
	}
}
